openstack/baremetal/v1/nodes/requests.go

```diff
@@ -301,3 +320,118 @@ func Delete(client *gophercloud.ServiceClient, id string) (r DeleteResult) {
 	_, r.Err = client.Delete(deleteURL(client, id), nil)
 	return
 }
+
+// A cleaning step has required keys ‘interface’ and ‘step’, and optional key ‘args’. If specified,
+// the value for ‘args’ is a keyword variable argument dictionary that is passed to the cleaning step
+// method.
+type CleanStep struct {
+	Interface string            `json:"interface,required"`
+	Step      string            `json:"step,required"`
+	Args      map[string]string `json:"args,omitempty"`
+}
+
+// ProvisionStateOptsBuilder allows extensions to add additional parameters to the
+// ChangeProvisionState request.
+type ProvisionStateOptsBuilder interface {
+	ToProvisionStateOpts() (map[string]interface{}, error)
+}
+
+// ProvisionStateOpts for a request to change a node's provision state.
+type ProvisionStateOpts struct {
+	Target         TargetProvisionState `json:"target,required"`
+	ConfigDrive    ConfigDriveBuilder   `json:"-"`
+	CleanSteps     []CleanStep          `json:"clean_steps,omitempty"`
+	RescuePassword string               `json:"rescue_password,omitempty"`
+}
+
+// ConfigDriveBuilder allows extensions to change how the configdrive is constructed.
+type ConfigDriveBuilder interface {
+	ToConfigDrive() (*string, error)
+}
+
+// A configdrive may be a path to a directory, or an image file
+type ConfigDrivePath string
+
+// A configdrive may also be a URL pointing to such an image
+type ConfigDriveValue string
```

Comment:
I'm not sure if this is the right way to handle it, but a config drive can be a URL or a base64 encoded gzipped image directly so there's two types and each handles serializing it for the API call differently.

Reply:
I'm working on accepting a dict as configdrive: https://storyboard.openstack.org/#!/story/2005083